### PR TITLE
gh-125522: Fix bare except in test_math.testTan

### PR DESCRIPTION
**Pull request by iritkatriel**
This is a simple rewrite of the test to avoid bare except and make the test's intention clearer

* Issue: gh-125522

### CODE REVIEW

File: Lib/test/test_math.py [hunk 1]
```diff
@@ -1899,14 +1899,22 @@ def testTan(self):
...
+        else:
+            self.assertTrue(math.isnan(tan))
+
+        try:
+            ntan = math.tan(NINF)
+        except ValueError:
+            pass
+        else:
+            self.assertTrue(math.isnan(ntan))
+
```
Comment (skirpichev) [inline Lib/test/test_math.py]: I concur that this is just a rewrite.  But maybe it's an opportunity to simplify test?

If libm's tan returns nan for non-nan input - the math_1() raises and exception:
https://github.com/python/cpython/blob/d08c7888229e78533648191dfe42e2d2d3ecea25/Modules/mathmodule.c#L932-L936

C stdandards also say if the implementation supports IEEE floating-point arithmetic (IEC 60559): if the argument is ±∞, NaN is returned and [FE_INVALID](https://en.cppreference.com/w/c/numeric/fenv/FE_exceptions) is raised.

```suggestion
        self.assertRaises(ValueError, math.tan, INF)
        self.assertRaises(ValueError, math.tan, NINF)
```

Comment (iritkatriel) [reply Lib/test/test_math.py]: What is the history behind the way the test is currently written?

Comment (skirpichev) [reply Lib/test/test_math.py]: This seems to be added in 53876d9cd8, but I don't see yet any glue why.

Comment (skirpichev) [reply Lib/test/test_math.py]: Maybe it *might* be relevant for some alternative implementation, that doesn't require support for IEEE 754 from platform and NaN.  @serhiy-storchaka 

If it worth saving, lets make it at least consistent with other similar tests (testSin/Cos):
```suggestion
        try:
            self.assertTrue(math.isnan(math.tan(INF)))
            self.assertTrue(math.isnan(math.tan(NINF)))
        except ValueError:
            self.assertRaises(ValueError, math.tan, INF)
            self.assertRaises(ValueError, math.tan, NINF)
```

The implementation will return ``nan`` for one infinity and an exception for other - seems to be too odd to accept.

Comment (iritkatriel) [reply Lib/test/test_math.py]: Yes, good point.

**Review by skirpichev — APPROVED** [review]

**Review by serhiy-storchaka — APPROVED** [review]
LGTM.

**Comment by skirpichev** [comment]
@serhiy-storchaka, what do you think on keeping only self.assertRaises?

**Comment by serhiy-storchaka** [comment]
I think that this test implies that NaN is a valid returning value. If we want to change this, open a separate issue. It should be discussed with Mark and Tim.

**Pull request merged**